flow.tcl

```diff
@@ -37,10 +37,11 @@ proc run_non_interactive_mode {args} {
 	run_routing
 
 	if { $::env(DIODE_INSERTION_STRATEGY) == 2 } {
-	    run_magic_antenna_check; # produces a report of violators; extraction!
+		run_antenna_check
 	    heal_antenna_violators; # modifies the routed DEF
 	}
-
+	run_antenna_check
```

Comment:
This isn't inside an else block, so you're running it twice

Maybe make a comment that this is intentional?

Reply:
The process goes as follows:
if Diode strategy 2 -> run the antenna checks generate the antenna report -> use the antenna report to replace fake diodes with real diodes (reducing the number of antenna violations)

Then, either way, run the antenna checks to generate the final report.